common/include/pcl/impl/point_types.hpp

```diff
@@ -362,7 +362,7 @@ namespace pcl
 
   struct _Intensity
   {
-    PCL_ADD_INTENSITY;
+    PCL_ADD_INTENSITY
```

Comment:
Personally I prefer removing the semicolon from the macro implementation to make the usage more uniform (no guessing if `;` is needed or not: it's always needed).

Reply:
I think I would actually go the other way and remove even more unnecessary semicolons, like after `PCL_ADD_POINT4D`. IMO it is clearer if the semicolon is directly after the definition of the union or struct, and some macros like `PCL_ADD_EIGEN_MAPS_POINT4D` do not need a semicolon at the end.

Reply:
I added another commit with my suggestion

Reply:
>  it is clearer if the semicolon is directly after the definition of the union or struct

If that's the approach, then I'd suggest renaming the macros to `PCL_ADD_STRUCT_X`.

> macros like `PCL_ADD_EIGEN_MAPS_POINT4D` do not need a semicolon at the end

I'd consider that to be a something not to do and not a motivating example. It's possible to force a semi-colon at the end using: `static_assert(true, "Macro needs to end with a semi-colon")`.

I've often seen people argue towards requiring a semi-colon because that implicitly says that the macro definition can't affect the line of code you'd be writing.

[Example 1](https://wiki.sei.cmu.edu/confluence/display/c/PRE11-C.+Do+not+conclude+macro+definitions+with+a+semicolon), [example 2](https://stackoverflow.com/questions/321143/good-programming-practices-for-macro-definitions-define-in-c)

Reply:
Hm, since this sparks a larger discussion than I anticipated, let's maybe focus on the includes and I will remove the semicolon-commits?